nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -103,9 +114,25 @@ class SimpleActionServer
       action_name_,
       std::bind(&SimpleActionServer::handle_goal, this, _1, _2),
       std::bind(&SimpleActionServer::handle_cancel, this, _1),
-      std::bind(&SimpleActionServer::handle_accepted, this, _1));
+      std::bind(&SimpleActionServer::handle_accepted, this, _1),
+      rcl_action_server_get_default_options(),
+      callback_group_);
+    if (spin_thread_) {
+      auto executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+      executor->add_callback_group(callback_group_, node_base_interface_);
+      executor_thread_ = std::make_unique<nav2_util::NodeThread>(executor);
+    }
   }
 
+  /**
+   * @brief A destructor
+   */
+  ~SimpleActionServer()
+  {
+    if (spin_thread_) {
+      executor_thread_.reset();
```

Comment:
I think you need to join the thread right? Or is that handled in the destructor of the node thread class? This is kind of redundant. The unique pointer will be destroyed when this destructor finishes.

Reply:
executor_thread_ is type of `NodeThread`, thread will be  joined in destructor of `NodeThread`, https://github.com/ros-planning/navigation2/blob/main/nav2_util/src/node_thread.cpp#L41-L45

Reply:
Right, so this code block (and the whole destructor) isn't really needed. Right?

Reply:
yes, i think executor_thread_ will auto destruct when SimpleActionServer destructs. but i refer some code like https://github.com/ros-planning/navigation2/blob/main/nav2_planner/src/planner_server.cpp#L70. 
anyway, i will remove it. i will add it again if need.